xbmc/cdrip/CDDARipJob.cpp

```diff
@@ -192,8 +192,8 @@ CEncoder* CCDDARipJob::SetupEncoder(CFile& reader)
     return NULL;
 
   // we have to set the tags before we init the Encoder
-  const std::string strTrack =
-      StringUtils::Format("{}", strtol(m_input.substr(13, m_input.size() - 13 - 5), nullptr, 10));
+  const std::string strTrack = StringUtils::Format(
```

Comment:
This looks odd. Why do we first convert a string to an integer and then format it back to a string? Is this on purpose to remove any non-digit characters? Otherwise `const auto strTrack = m_input.substr(13, m_input.size() - 13 - 5);` would be enough.

Reply:
I'm really not sure 😕

Reply:
CD ripping in Kodi is ancient. My guess would be to remove things like leading zeros or non-digits, but I don't know either despite it being music related. 

I actually think Kodi should not be a ripper at all, there is so many better software tools for that out there. We want users to fully tag their music, even best (for efficient scraping of extra art etc.) with Musicbranz ids, and Kodi can't do that just offers basic CDDA (track naming based). Sorry a digression!